yt/yt/tests/integration/node/test_job_proxy.py

```diff
@@ -434,16 +448,131 @@ def test_separate_directory(self):
+                "enable": False,
+                "analysis_period": 100,
+                # Cleanup all operations
+                "hard_retained_operation_count": 0,
+                "clean_delay": 0,
+            },
+            "enable_job_reporter": True,
+            "enable_job_spec_reporter": True,
+            "enable_job_stderr_reporter": True,
+        },
+    }
+
+    DELTA_DYNAMIC_NODE_CONFIG = {
+        "%true": {
+            "exec_node": {
+                "job_reporter": {
+                    "reporting_period": 10,
+                    "min_repeat_delay": 10,
+                    "max_repeat_delay": 10,
+                },
+            },
+        },
+    }
+
+    def setup_method(self, method):
+        super(TestDumpJobProxyLog, self).setup_method(method)
+        sync_create_cells(1)
+        init_operations_archive.create_tables_latest_version(
+            self.Env.create_native_client(), override_tablet_cell_bundle="default"
+        )
+
+    def teardown_method(self, method):
+        remove("//sys/operations_archive", force=True)
+        super(TestDumpJobProxyLog, self).teardown_method(method)
+
+    def validate_dumped_logs(self, path, job_id):
+        for line in read_file(path).decode("utf-8").split("\n"):
+            if "Job spec received" in line:
+                assert job_id in line
+
+    @authors("tagirhamitov")
+    def test_dump_for_running_job(self):
+        path = "//tmp/job_proxy.log"
+        create("file", path)
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"))
+        job_id = wait_breakpoint()[0]
+        dump_job_proxy_log(job_id, op.id, path)
+        release_breakpoint()
+        op.track()
+
+        self.validate_dumped_logs(path, job_id)
+
+    @authors("tagirhamitov")
+    def test_dump_for_finished_job(self):
+        path = "//tmp/job_proxy.log"
+        create("file", path)
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"))
+        job_id = wait_breakpoint()[0]
+        release_breakpoint()
+        op.track()
+
+        dump_job_proxy_log(job_id, op.id, path)
+        self.validate_dumped_logs(path, job_id)
+
+    @authors("tagirhamitov")
+    def test_dump_for_missing_job(self):
+        update_nodes_dynamic_config({
+            "exec_node": {
+                "job_controller": {
+                    "job_proxy_log_manager": {
+                        "logs_storage_period": "0s",
+                    },
+                }
+            }
+        })
+        path = "//tmp/job_proxy.log"
+        create("file", path)
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"))
+        job_id = wait_breakpoint()[0]
+        release_breakpoint()
+        op.track()
+
+        log_path = os.path.join(self.path_to_run, log_path_from_job_id(job_id))
+        wait(lambda: not os.path.exists(log_path))
+
+        with pytest.raises(YtError) as e:
+            dump_job_proxy_log(job_id, op.id, path)
+            assert "Logs not found for JobId" in e
```

Comment:
I don't think that this line is reached.

The correct check should look like this: https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/tests/integration/controller/test_map_operation.py#L2167